assignments/array-methods.js

```diff
@@ -58,28 +58,65 @@ const runners = [
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs. Combine both the first and last names and populate a new array called `fullNames`. This array will contain just strings.
 let fullNames = [];
+runners.forEach(function(arr){
+  fullNames.push(arr.first_name + " " + arr.last_name);
+});
 console.log(fullNames);
 
+
+
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runners' first names in uppercase because the director BECAME DRUNK WITH POWER. Populate an array called `firstNamesAllCaps`. This array will contain just strings.
 let firstNamesAllCaps = [];
+runners.map((data) => {
+  return firstNamesAllCaps.push(data.first_name.toUpperCase());
+})
 console.log(firstNamesAllCaps);
 
+
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue. We need a filtered version of the runners array, containing only those runners with large sized shirts so they can choose a different size. This will be an array of objects.
 let runnersLargeSizeShirt = [];
+runnersLargeSizeShirt = runners.filter(function largeOnly (data) {
+return data.shirt_size.includes("L")});
 console.log(runnersLargeSizeShirt);
+//I just realised I may have done the above question wrong. I thought that 'large sized shirts' meant any shirt that was size L and above. To fix this, I would return shirt size that === "L".
```

Comment:
Yeah this could have done a little bit easier with the help of [filter method](https://javascript.info/array-methods#filter). With your solution, you're getting all the shirt sizes that includes letter "L". Make your life a bit easier by choosing a proper method to work with. Here's a simple solution for this part:
`let largeShirts = runners.filter(runner => runner.shirt_size === "L");
console.log(largeShirts);`